cannon/mipsevm/state.go

```diff
@@ -60,3 +65,37 @@ func (s *State) EncodeWitness() []byte {
 	}
 	return out
 }
+
+type StateWitness []byte
+
+const (
+	VMStatusValid      = 0
+	VMStatusInvalid    = 1
+	VMStatusPanic      = 2
+	VMStatusUnfinished = 3
+)
+
+func (sw StateWitness) StateHash() common.Hash {
+	hash := crypto.Keccak256Hash(sw)
+	offset := 32*2 + 4*6
+	exited := sw[offset]
+	exitCode := sw[offset+1]
+	status := vmStatus(exited == 1, exitCode)
+	hash[0] = status
+	return hash
+}
+
+func vmStatus(exited bool, exitCode uint8) uint8 {
+	if exited {
```

Comment:
nit: Invert this condition and return early?

Reply:
++

```golang
func vmStatus(exited bool, exitCode uint8) uint8 {
	if !exited {
		return VMStatusUnfinished
    }

    switch exitCode {
	case 0:
		return VMStatusValid
	case 1:
		return VMStatusInvalid
	default:
		return VMStatusPanic
	}
}
```